### PR TITLE
feat:add function to list commit and add additional information to co…

### PR DESCRIPTION
**Pull request by XiaoBoya**
# Changes

add function for list commit and add additional information to commit

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [√] [`spec` feat/add-doc-for-listcommit](https://github.com/katanomi/spec/pull/82) included
- [√] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [√] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [√] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [√] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

API changes

For pull requests with a release note:

```release-note
add function list commit and add additional information to commit
```

### CODE REVIEW

File: apis/meta/v1alpha1/gitcommit_types.go [hunk 1]
```diff
@@ -43,13 +44,16 @@ type GitCommitBasicInfo struct {
 // GitCommitInfo github support field is SHA & web_URL
 type GitCommitInfo struct {
 	// SHA commit's sha
-	SHA      *string     `json:"sha,omitempty"`
+	SHA *string `json:"sha,omitempty"`
+	// TODO remove this field, because is same as metadata.creationTimeStamp
 	CreateAt metav1.Time `json:"createAt"`
+	// Address for commit url for code repository web server
```
Comment (danielfbm) [inline apis/meta/v1alpha1/gitcommit_types.go:50]: CreateAt 有个type，应该是 `CreatedAt`

File: apis/meta/v1alpha1/gitoption_types.go [hunk 2]
```diff
@@ -24,6 +28,14 @@ type GitRepoFileOption struct {
 	Path string `json:"path"`
 }
 
+// GitCommitListOption option for list commit
+type GitCommitListOption struct {
+	GitRepo
+	Ref   string   `json:"ref"`
+	Since *v1.Time `json:"since,omitempty"`
+	Until *v1.Time `json:"util,omitempty"`
```
Comment (danielfbm) [inline apis/meta/v1alpha1/gitoption_types.go:36]: 加注释？

File: plugin/route/gitcommit.go [hunk 3]
```diff
@@ -69,3 +74,71 @@ func (a *gitCommitGetter) GetCommit(request *restful.Request, response *restful.
 	}
 	response.WriteHeaderAndEntity(http.StatusOK, commitObject)
 }
+
+func HandleTimeParamInQuery(param string) (res *v1.Time, err error) {
```
Comment (danielfbm) [inline plugin/route/gitcommit.go:78]: 注释

File: plugin/client/gitcommit.go [hunk 4]
```diff
@@ -63,3 +65,25 @@ func (g *gitCommit) Get(ctx context.Context, baseURL *duckv1.Addressable, option
...
+		query["since"] = option.Since.Format(time.RFC3339)
+	}
+	if option.Until != nil {
+		query["until"] = option.Until.Format(time.RFC3339)
+	}
+	options = append(options, QueryOpts(query))
+	uri := fmt.Sprintf("projects/%s/coderepositories/%s/commits", option.Project, option.Repository)
+	if err := g.client.Get(ctx, baseURL, uri, options...); err != nil {
+		return nil, err
+	}
```
Comment (danielfbm) [inline plugin/client/gitcommit.go:87]: 这个问题跟科燃的PR是一个问题，你们达成一致一个方案统一修复吧

现在新添加的插件函数/API在这里调度会导致 404 错误（路由不能存在），用户不好判断到底是代码仓库不存在，还是啥问题，你们一起考虑一个方案哈

File: plugin/route/options.go [hunk 5]
```diff
@@ -62,7 +70,7 @@ func HandleSortQuery(ctx context.Context, sortBy string) []metav1alpha1.SortOpti
 	sortList := []metav1alpha1.SortOptions{}
 	sortInfoList := strings.Split(sortBy, ",")
 	if len(sortInfoList)%2 != 0 {
-		logger.Errorw("It is singular after cutting, so it is impossible to obtain valid sorting conditions.Ignore sortBy!!!", "sortBy", url.QueryEscape(sortBy))
+		logger.Errorw("It is singular after cutting, so it is impossible to obtain valid sorting conditions.Ignore sortBy!!!", SortQueryKey, url.QueryEscape(sortBy))
```
Comment (danielfbm) [inline plugin/route/options.go:73]: `"It is singular after cutting, so it is impossible to obtain valid sorting conditions.Ignore sortBy!!!"` > `"sortBy is expected to be in pairs ignoring sort by..."`

然后在values里面加 sortInfoList 的len

**Review by danielfbm — CHANGES_REQUESTED** [review]

File: plugin/client/gitcommit.go (hunk 4 above)
Comment (XiaoBoya) [reply plugin/client/gitcommit.go:87]: https://github.com/katanomi/spec/pull/92

**Review by danielfbm — APPROVED** [review]

**Review by danielfbm — APPROVED** [review]

**Review by danielfbm — APPROVED** [review]

**Pull request merged**